sktime/tests/test_switch.py

```diff
@@ -0,0 +1,43 @@
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Switch utility for determining whether tests for a class should be run or not."""
+
+__author__ = ["fkiraly"]
+
+
+def run_test_for_class(cls):
+    """Check if test should run for a class.
+
+    This checks the following conditions:
+
+    * if required dependencies are not present, does not run the test
+    * if ONLY_CHANGED_MODULES setting is on, runs the test
+      if and only if the class has changed according ot is_class_changed
+
+    otherwise, always runs the test.
+
+    Parameters
+    ----------
+    cls : class or list of class
+        class for which to determine whether it should be tested
+
+    Returns
+    -------
+    bool : True if class should be tested, False otherwise
+        if cls was a list, is True iff True for at least one of the classes in the list
+    """
+    if isinstance(cls, list):
+        return any(run_test_for_class(x) for x in cls)
+
+    from sktime.tests.test_all_estimators import ONLY_CHANGED_MODULES
+    from sktime.utils.git_diff import is_class_changed
+    from sktime.utils.validation._dependencies import _check_estimator_deps
```

Comment:
Can these not be at module level?

If I am not wrong, these don't depend on any soft dependencies, and top level imports are usually recommended by linters, style guides, etc.

Reply:
what do you mean by "these" - the `validation._dependencies`?

Yes, I suppose so, but this is "how it has always been".
In `skbase` it's moved to the top already, and in `sktime` it would be a fiddly refactor.

Reply:
I just meant to move the 3 import lines from inside the function definition to between module docstring and `__author__`. If it's complicated or likely to have side-effects, please ignore.

Reply:
yes, worried about side effects in the module. Probably over-worrying though.